### PR TITLE
[OEP-21] Migrate DEPR tracking from Jira to GitHub Issues

### PR DESCRIPTION
**Pull request by sarina**
tCRIL is migrating ticket tracking from Jira to GH Issues. This set of changes describes how to appropriately create a GitHub issue for DEPR of something, and where and how to find the appropriate GitHub project board.

When this proposal is accepted, we will simultaneously publish the update and migrate all existing, in-progress DEPR Jira tickets to GH Issues.

### CODE REVIEW

File: oeps/processes/oep-0021-proc-deprecation.rst [hunk 1]
```diff
@@ -91,6 +109,9 @@ code, that is, code that may be superseded by a new implementation or may be dee
 longer necessary or supported. Following this process guarantees alignment
 across all stakeholders and full execution.
 
+**For code review: "full execution" in the above line makes no sense to me. What is that
+trying to convey and how could I reword it?**
```
Comment (kdmccormick) [inline oeps/processes/oep-0021-proc-deprecation.rst:113]: perhaps "a path towards complete removal of all unused code"?

File: oeps/processes/oep-0021-proc-deprecation.rst [hunk 2]
```diff
@@ -268,6 +311,8 @@ guidance. In some cases, the proposal may need to be *Abandoned* entirely.
 
 .. _edX Architecture Team: https://openedx.atlassian.net/wiki/spaces/AC/pages/439353453/Architecture+Team
 
+**For Code Review: This link is not publicly available. Where should we direct people?**
```
Comment (sarina) [inline oeps/processes/oep-0021-proc-deprecation.rst:314]: Code reviewers: Please advise

File: oeps/processes/oep-0021-proc-deprecation.rst [hunk 3]
```diff
@@ -286,25 +331,28 @@ Here are some common ways to mark a technology as deprecated:
...
 * Feature toggles - Set the “Expiration Date” as described in OEP-17_.
 
-* Github repo - See `OEP-14 <https://open-edx-proposals.readthedocs.io/en/latest/oep-0014-proc-archive-repos.html>`_
+* GitHub repo - See `OEP-14 Archiving Open edX GitHub Repositories`_.
+
+* XBlock - For edx.org specifically, see `Deprecating xBlock for the edX website`_ (internal edX use only).
 
-* XBlock - For edx.org specifically, see `Deprecating xBlock for the edX website`_.
+**For Code Review: The above link is specific to edX and not externally visible. Should it
+really be part of an OEP? If so, should other community members get to insert their own process into OEPs as well?**
```
Comment (sarina) [inline oeps/processes/oep-0021-proc-deprecation.rst:351]: Code reviewers: Please advise

File: oeps/processes/oep-0021-proc-deprecation.rst (hunk 2 above)
Comment (dianakhuang) [reply oeps/processes/oep-0021-proc-deprecation.rst:314]: Oh wow, that's really old. I would maybe suggest the deprecation working group page (which I believe should be public): https://openedx.atlassian.net/wiki/spaces/AC/pages/825983190/Deprecation+Working+Group

File: oeps/processes/oep-0021-proc-deprecation.rst (hunk 3 above)
Comment (dianakhuang) [reply oeps/processes/oep-0021-proc-deprecation.rst:351]: I'm okay removing this link and moving it to just internal 2U/edX documentation.

File: oeps/processes/oep-0021-proc-deprecation.rst [hunk 4]
```diff
@@ -7,7 +7,7 @@ OEP-21: Deprecation and Removal
 +-----------------+--------------------------------------------------------+
 | Title           | Deprecation and Removal                                |
 +-----------------+--------------------------------------------------------+
-| Last Modified   | 2020-12-09                                             |
+| Last Modified   | 2022-01-12                                             |
 +-----------------+--------------------------------------------------------+
 | Authors         | Greg Sham <gsham@edx.org>,                             |
 |                 | Nimisha Asthagiri <nimisha@edx.org>                    |
```
Comment (dianakhuang) [inline oeps/processes/oep-0021-proc-deprecation.rst:13]: Do you also want to be an author? I don't mind either way.

**Review by dianakhuang — APPROVED** [review]
It looks good to me, and I've added my comments on your existing questions. I'll make sure that the DEPR group knows about these changes and hopefully can help out with any migration that needs to happen.

File: oeps/processes/oep-0021-proc-deprecation.rst (hunk 3 above)
Comment (sarina) [reply oeps/processes/oep-0021-proc-deprecation.rst]: Done. Will you take care of finding the proper place for it?

**Comment by nedbat** [comment]
@sarina nice work updating the diagrams!!

**Comment by nedbat** [comment]
I don't see a good reason to use GitHub issues for DEPR if the repo isn't using issues to track other things.  Making a template that says "You can only create DEPR issues here" just seems like opening the door and then trying to keep everyone out.

**Comment by kdmccormick** [comment]
@nedbat We'd like everything in GH issues eventually. This is a way to tiptoe towards that and see how things go.

To use your analogy, we're cracking the door open and peering inside before telling everyone to go through.

**Comment by nedbat** [comment]
But we know there is a large impediment just beyond this tiptoe: how to get maintainers who don't already use GitHub issue to attend to them.

The usual reason to write an issue in a particular repo is so that the people watching that repo will see the issue.  Who is the audience for these DEPR issues? Are those people watching the repos' issues?

At the very least, we need someone for each repo to agree to watch the issues in that repo.

**Comment by feanil** [comment]
@nedbat given that we're only going to be introducing DEPR tickets as a valid type for now, I think this work can proceed. The DEPR working group would be monitoring all repos for DEPR tickets and managing them.  I do agree that it will not be a good test of how other kinds of issues will work in the future.  ie. If Bugs and Feature requests live in GitHub issues, we will need a more sustained effort from repo maintainers to be triaging and managing these backlogs.  I'm not sure we have agreement/resources to do that today and before turning those things on we should get proper buy in from repo maintainers(which I think will be codified in an upcoming OEP).

**Comment by sarina** [comment]
Hi all,

I have made a first attempt to address the following constraints:

1. We would like to create DEPR tickets in GH Issues
2. In order to do so, we will wish to gradually turn on GH Issues across multiple repos
3. We would like to try to head off potential spam, as well as questions/bug reports which ought to go in the forums

Further, I would like to avoid over-engineering a solution. I would like to get consensus that this MVP looks good enough, and if needed, we can tweak it if there is too much being created than we can keep up with.

With that, are there any thoughts on how this looks/ how it might address concerns around opening up GitHub issues on more repos? [Example new issue workflow for repo only accepting DEPR tickets](https://github.com/openedx/tcril-onboarding/issues/new/choose)

Making blank issues is disabled. You’re clearly directed to the forums. And the DEPR template’s first line says “this is only for DEPR”. I _could_ make a checkbox that asks people to ack that they understand the purpose of the ticket, but to me that feels oddly unwelcoming and punitive and I'd consider only post-MVP, if we think we need it.

**Comment by kdmccormick** [comment]
@sarina Nice. I think your template will be helpful and effective, both for folks who do mean to file a DEPR and for those who are trying to report something else.

If I were to suggest one thing, you might expand:
> Please ask all questions and report all bugs here.

to:
> Please ask all questions, suggest all enhancements, and report all bugs here.

in order to make it abundantly clear that GitHub issues is not (yet?) the place for enhancement requests. But please do not take this suggestion as blocking.

File: oeps/processes/oep-0021-proc-deprecation.rst (hunk 3 above)
Comment (nedbat) [reply oeps/processes/oep-0021-proc-deprecation.rst]: The page is now here: https://openedx.atlassian.net/wiki/spaces/COMM/pages/151912534/Proposed+Github+Deprecation+Process

**Comment by dianakhuang** [comment]
@sarina the example flow you have looks great! 👍

**Comment by sarina** [comment]
@kdmccormick thanks for the feedback, I agree that's clearer. Changed!

File: oeps/processes/oep-0021-proc-deprecation.rst (hunk 3 above)
Comment (sarina) [reply oeps/processes/oep-0021-proc-deprecation.rst]: I don't see that referenced anywhere in this doc (https://open-edx-proposals.readthedocs.io/en/latest/processes/oep-0021-proc-deprecation.html)

The comment I wrote referred to this doc: https://openedx.atlassian.net/wiki/spaces/ENG/pages/723550424/Deprecating+and+Disabling+an+XBlock+for+the+edX+website

I could see keepi ng that in as a reference to other site operators, but the page would need to be public.

**Pull request merged**